lib/ansible/plugins/action/package.py

```diff
@@ -45,14 +47,22 @@ def run(self, tmp=None, task_vars=None):
         if module == 'auto':
             try:
                 if self._task.delegate_to:  # if we delegate, we should use delegated host's facts
-                    module = self._templar.template("{{hostvars['%s']['ansible_facts']['pkg_mgr']}}" % self._task.delegate_to)
+                    module = C.config.get_config_value('PACKAGE_MANAGER_OVERRIDE', variables=combine_vars(self._task.vars, task_vars.get('delegated_vars', {})))
+                    if module is None:
+                        dh = '%s' % self._task.delegate_to
+                        module = self._templar.template(
+                            f"{{hostvars['{dh}']['ansible_local']['overrides']]['pkg_mgr']|default(hostvars['{dh}']['ansible_facts']['pkg_mgr'])}}"
+                        )
                 else:
-                    module = self._templar.template('{{ansible_facts.pkg_mgr}}')
+                    module = C.config.get_config_value('PACKAGE_MANAGER_OVERRIDE', variables=task_vars)
+                    if module is None:
+                        module = self._templar.template("{{ansible_local['overrides']['pkg_mgr']|default(ansible_facts['pkg_mgr'])}}")
             except Exception:
                 pass  # could not get it from template!
 
         try:
             if module == 'auto':
```

Comment:
Seems like this can never run now- won't the `auto` value be trashed with a `None` by the config calls above?